core/src/main/scala/org/ensime/core/Analyzer.scala

```diff
@@ -185,18 +185,29 @@ class Analyzer(
       sender ! VoidResponse
     case TypecheckModule(moduleName) =>
       //consider the case of a project with no modules
-      config.modules get (moduleName) foreach {
-        module =>
-          val files: Set[SourceFileInfo] = module.scalaSourceFiles.map(s => SourceFileInfo(EnsimeFile(s), None, None))(breakOut)
-          sender ! scalaCompiler.handleReloadFiles(files)
+      val responses: List[RpcResponse] = for (module <- (config.modules filterKeys (_.project == moduleName)).values.toList) yield {
+        val files: Set[SourceFileInfo] = module.scalaSourceFiles.map(s => SourceFileInfo(EnsimeFile(s), None, None))(breakOut)
+        scalaCompiler.handleReloadFiles(files)
+      }
+      val (_, errors) = responses.partition(_ == VoidResponse)
+      if (errors.isEmpty)
+        sender ! VoidResponse
+      else {
+        val accumulatedErrorString = (errors foldLeft "") { (allErrors, err) =>
+          err match {
+            case EnsimeServerError(desc) => allErrors + "\n" + desc
+            case _ => allErrors
+          }
+        }.tail
+        sender ! EnsimeServerError(accumulatedErrorString)
       }
     case UnloadModuleReq(moduleName) =>
-      config.modules get (moduleName) foreach {
+      (config.modules filterKeys (_.project == moduleName)).values foreach {
         module =>
           val files = module.scalaSourceFiles.toList
           files.foreach(scalaCompiler.askRemoveDeleted)
-          sender ! VoidResponse
       }
+      sender ! VoidResponse
```

Comment:
This is kind of a hack though. Currently if we received a request and we had no module named `moduleName`, we wouldn't send back a Void Response. Otherwise, it's same.
Will that be a problem? If so, we can check if we get some EnsimeProject with `module.id.project` equal to moduleName and send back a Void Response only then.
But does this really matter?